### PR TITLE
Adding Randomized Latin Hypercube class as long as tests and document…

### LINKED ISSUE #573: Add Randomized Latin Hypercube Sampling (RLHS) discrete distribution
## Motivation

QMCPy provides several low-discrepancy constructions, but does not appear to provide a Randomized Latin Hypercube Sampling (RLHS) distribution.

RLHS is a variance reduction technique that ensures uniform coverage of each dimension while preserving the unbiasedness and convergence properties of standard uniform random sampling.

## Proposed feature

Add a Randomized Latin Hypercube Sampling (RLHS) discrete distribution following the existing QMCPy distribution interfaces and conventions.

The construction is based on

$$
X_i^j = \frac{\Pi_j(i) - U_i^j}{n}, \quad i = 1, \dots, n,\; j = 1, \dots, d,
$$

where $\Pi_j$ are independent uniform random permutations of $\{1, \dots, n\}$, and $U_i^j \sim \mathcal{U}([0, 1[)$ are i.i.d. and independent of the permutations.

Each point satisfies $X_i \sim \mathcal{U}([0, 1[^d)$, so all results for uniform random sequences apply (unbiased estimators, convergence rate, etc.). In addition, RLHS provides:

- Better space-filling properties than pure random sampling.
- Exact stratification in each dimension, ensuring that every coordinate stratum is represented exactly once per dimension.
- Reduced variance for integrands that are monotonic or additive.

## Proposed implementation

- Add the new distribution to the appropriate QMCPy module.
- Follow the existing distribution API.
- Add unit tests for the mathematical construction and edge cases.
- Add documentation and examples.
- Add references to the relevant literature.

I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by Samuel-Vangu**
## Summary
- Issue: Add Randomized Latin Hypercube Sampling (RLHS) discrete distribution #573
- Algorithmic or API impact: Adds a new `RandomizedLHS` discrete distribution class implementing Latin Hypercube Sampling (stratified random permutation + within-stratum jitter). Purely additive: no existing class, method, or public API signature is modified. Exported via `qmcpy/discrete_distribution/__init__.py` alongside the other discrete distributions.
- Commands run:
  - `python -m pytest test/test_discrete_distribs.py -v`
  - `python -m pytest test/ -q` (full suite)
  - `python -m doctest qmcpy/discrete_distribution/randomized_lhs.py -v`
  - `mkdocs build --strict`
  - `make doc`

## AI Assistance
- [ ] No substantive AI assistance was used for this PR.
- [x] AI assistance substantively affected this PR, and I describe that use below.

AI tools and affected areas:
- Used an AI assistant (Claude) to review my initial implementation of `RandomizedLHS` and catch bugs: an incorrect override of `gen_samples` instead of `_gen_samples`, a reference to a non-existent `self.n_samples` attribute, an incorrect axis transpose, and a missing `__repr__` override causing a `TypeError` in the shared test suite.
- Used AI assistance to draft the unit tests in `test/test_discrete_distribs.py` (shape checks, stratification property, reproducibility, warnings, `spawn` behavior, and hardcoded regression values for a fixed seed).
- Used AI assistance to improve the class docstring (structure, `Examples` block, and bibliographic references), following the style of the existing `Halton` docstring.

- [x] Independent verification performed.

I ran the full test suite and doctests myself after each change (not just the AI's proposed diff) and confirmed all pass locally (36/36 in `test_discrete_distribs.py`, full suite green, doctests pass). I independently verified the five bibliographic references added to the docstring (title, journal, volume, pages, DOI) via web search rather than trusting them as given. I also manually confirmed the stratification and shape guarantees hold by inspecting generated samples directly.

## Checklist
- [x] I linked the relevant issue or explained why none was needed.
- [x] I added or updated tests, docs, notebooks, or explained why they were not needed.
- [x] I verified any equations, citations, benchmarks, numerical claims, or external references changed in this PR.
- [x] I reviewed AI-assisted content for licensing, provenance, attribution, confidentiality, and security concerns.
- [x] I described any CI, dependency, notebook runtime, or generated artifact changes if applicable.

No new dependencies are introduced by this PR itself (only existing `docs`/`test` extras were used locally to build documentation and run tests). Regenerated UML diagrams under `docs/api/umls/` reflect the new class in the discrete distribution hierarchy.

Closes #573

### CODE REVIEW

**Comment by Samuel-Vangu** [comment]
Hi @fjhickernell  @alegresor , I pushed some of the changes we talked about for LatinHypercube:

- Renamed the class from `RandomizedLHS` to `LatinHypercube`.
- Added a `randomize` parameter (defaults to `True`); when `False`, it performs a
  centered (non-randomized) LHS, placing each point at the center of its stratum
  instead of a random position within it.
- Updated everything else to match (docstrings, tests, etc.).

Let me know if I'm missing anything else from what we discussed, or if you'd like
me to adjust anything further.

File: qmcpy/discrete_distribution/latin_hypercube.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+                f"randomize must be one of 'TRUE', 'FALSE', 'NONE', or 'NO' (case-insensitive), got {randomize!r}."
+            )
+
+    def _gen_samples(
+            self, n=None, n_min=None, n_max=None, return_binary=False, warn=True
+        ):
+        r"""..."""  # (inchangee)
+        if return_binary:
+            raise ParameterError("LatinHypercube does not support return_binary=True")
+        if warn:
```
Comment (alegresor) [inline qmcpy/discrete_distribution/latin_hypercube.py]: only throw this warning if `n_min>0`

File: qmcpy/discrete_distribution/latin_hypercube.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        array([[0.125, 0.375],
+               [0.625, 0.875],
+               [0.375, 0.625],
+               [0.875, 0.125]])
+
+    Parameters
+    ----------
+    dimension (int): Dimension of the samples.
+
+    replications (Union[None, int]): Number of independent randomizations. This is implemented only for API consistency; equivalent to reshaping the samples.
```
Comment (alegresor) [inline qmcpy/discrete_distribution/latin_hypercube.py]: Not sure what is meant by "equivalent to reshaping the samples". I don't think generating with `replications=3` and `n=8` is the same as reshaping samples with `replications=None` and `n=3*8`

**Review by alegresor — CHANGES_REQUESTED** [review]
@Samuel-Vangu, looks good! Few minor suggestions below. Please re-request a review once you get a chance to change.

**Comment by alegresor** [comment]
Closing for #582
